setup.cfg

```diff
@@ -36,16 +44,19 @@ zip_safe = False
 packages = find:
 install_requires =
   matplotlib
-  numpy<1.20
-  pandas<0.25
-  pysat >=3.0
+  numpy
+  pandas
+  pysat>=3.0.2
+  xarray<2022.06.0
```

Comment:
There shouldn't be any version caps here

Reply:
the version limits aren't applied, or checked, during installation without specifying them here

Reply:
From docs on install_requires,  https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/, 
```
Additionally, it’s best practice to indicate any known lower or upper bounds.

```
The root version requirement on xarray should be placed on pysat, and re-released, so that the environments here work more out of the box.

Reply:
From your link:

```
Lastly, it’s important to understand that install_requires is a listing of “Abstract” requirements, i.e just names and version restrictions that don’t determine where the dependencies will be fulfilled from (i.e. from what index or source). The where (i.e. how they are to be made “Concrete”) is to be determined at install time using [pip](https://packaging.python.org/en/latest/key_projects/#pip) options. [1](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#id4)

[Requirements files](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#id6)
[Requirements Files](https://pip.pypa.io/en/latest/user_guide/#requirements-files) described most simply, are just a list of [pip install](https://pip.pypa.io/en/latest/cli/pip_install/#pip-install) arguments placed into a file.

Whereas install_requires defines the dependencies for a single project, [Requirements Files](https://pip.pypa.io/en/latest/user_guide/#requirements-files) are often used to define the requirements for a complete Python environment.
```

While the limits don't appear to be breaking the setup the way that they used to, it still seems cleaner to use the requirements file for this.  Might be best to discuss our standards at the next meeting.

Reply:
> From your link:
> 
> > Lastly, it’s important to understand that install_requires is a listing of “Abstract” requirements, i.e just names and version restrictions

> While the limits don't appear to be breaking the setup the way that they used to, it still seems cleaner to use the requirements file for this. Might be best to discuss our standards at the next meeting.


I'll put it on the agenda for Wed. though it is possible I may not be available. Government depending.

Reply:
I checked the linked discussion from pysatSpaceWeather: https://github.com/pypa/setuptools/issues/1951

This particular comment notes the anti-pattern is loading requirements.txt into install_requires. Further notes that the biggest issue with loading requirements.txt is the potential for pinned versions set with a `==`. Notes a compromise would be to allow requirements.txt to be loaded into install_requires as long as pinned versions aren't used.
https://github.com/pypa/setuptools/issues/1951#issuecomment-569530794


The python docs  https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/ say
```
install_requires is a [setuptools](https://packaging.python.org/en/latest/key_projects/#setuptools) setup.py keyword that should be used to specify what a project minimally needs to run correctly. When the project is installed by [pip](https://packaging.python.org/en/latest/key_projects/#pip), this is the specification that is used to install its dependencies.
```
I checked to see what was above this doc for greater context to see if this install_requires discussion was limited to a particular sub-situation but didn't see anything.

From what I've read setting minimal version requirements in install_requires in setup.cfg for pip installs is the correct thing to do. pip uses the information from setup to determine dependencies. For user pip installs of pysatPenumbra packages to work reliably we should specify a minimum pysat version in setup, 3.0.4, when that version is released. A similar condition would also be in requirements.txt, but strictly speaking the requirements and setup restrictions are there for different reasons thus may not always be the same. requirements.txt is not loaded into install_requires.

I'm not opposed to using a .toml file but I don't understand why that is better.

Reply:
I think it comes down to the idea that the format of a cfg is loosely defined.  See https://peps.python.org/pep-0518/#sticking-with-setup-cfg for why toml was recommended above cfg